### PR TITLE
Feat/more complex functions5

### PR DESCRIPTION
**Pull request by jtlap**

### CODE REVIEW

File: include/eve/module/complex/regular/detail/pow.hpp [hunk 1]
```diff
@@ -73,5 +74,40 @@ namespace eve
...
+    EVE_FORCEINLINE auto complex_binary_dispatch( eve::tag::pow_abs_
+                                                , Z1 const& z1, Z2 const& z2
+                                                ) noexcept
+    {
+      //     requires(floating_value<element_type_t<Z2>> || is_complex_v<Z2>)
+      if constexpr(is_complex_v<Z2>)
+        return pow(sqr_abs(z1), z2*half(eve::as(eve::real(z2))));
+      else
+      {
+//         std::cout << tts::typename_<Z1 >  << std::endl;
```
Comment (jfalcou) [inline include/eve/module/complex/regular/detail/pow.hpp:88]: remove comments

File: include/eve/module/complex/regular/detail/pow.hpp [hunk 2]
```diff
@@ -73,5 +74,40 @@ namespace eve
...
+//         std::cout << tts::typename_<decltype(sqr_abs(z1))> << std::endl;
+//         std::cout << tts::typename_<Z2 >  << std::endl;
+        return Z1{pow(sqr_abs(z1), z2*half(eve::as(eve::real(z1)))), 0};
+        //       return z1+z2;
```
Comment (jfalcou) [inline include/eve/module/complex/regular/detail/pow.hpp:92]: remove comments

File: include/eve/module/complex/regular/detail/arithmetic.hpp [hunk 3]
```diff
@@ -18,11 +18,17 @@ namespace eve::detail
...
-  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::trunc_,   Z const& z) noexcept { return Z{trunc(real(z)), trunc(imag(z))}; }
-  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::frac_,    Z const& z) noexcept { return Z{frac(real(z)), frac(imag(z))}; }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::trunc_,   Z const& z) noexcept { return Z{trunc(real(z)), trunc(imag(z))};     }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::frac_,    Z const& z) noexcept { return Z{frac(real(z)), frac(imag(z))};       }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::minus_,   Z const& z) noexcept { return Z{minus(real(z)), minus(imag(z))};     }
+  template<typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::plus_,    Z const& z) noexcept { return z;                                     }
+
+  template<decorator D, typename Z> EVE_FORCEINLINE auto complex_unary_dispatch(eve::tag::minus_, D const &, Z const& z) noexcept { return minus(z);        }
```
Comment (jfalcou) [inline include/eve/module/complex/regular/detail/arithmetic.hpp:29]: Why do we need those ?

File: include/eve/module/math/regular/impl/gd.hpp [hunk 4]
```diff
@@ -13,7 +13,7 @@
 namespace eve::detail
 {
 
-template<floating_real_value T>
+template<value T>
```
Comment (jfalcou) [inline include/eve/module/math/regular/impl/gd.hpp:16]: this means wide<int> is now a valid parameters. Do you need additional requirements ??

File: include/eve/module/complex/regular/detail/pow.hpp (hunk 2 above)
Comment (jtlap) [reply include/eve/module/complex/regular/detail/pow.hpp:92]: done

File: include/eve/module/complex/regular/detail/arithmetic.hpp (hunk 3 above)
Comment (jtlap) [reply include/eve/module/complex/regular/detail/arithmetic.hpp:29]: because minus is done for floating with a bitwise_xor

Comment (jtlap) [reply include/eve/module/complex/regular/detail/arithmetic.hpp:29]: but you are right I suppressed the plus and the decorated ones

File: include/eve/module/math/regular/impl/gd.hpp (hunk 4 above)
Comment (jtlap) [reply include/eve/module/math/regular/impl/gd.hpp:16]: there is in fact a small pb. The "real" formula is not correct for the whole complex plane so I will put back floating and add a correct complex version for gd and invgd

**Pull request merged**